### ISSUE TITLE
Possibility to apply multiple conditions on a single element of a one to many relation

### ISSUE THREAD

**Issue by tschodo** [realm/realm-java#1991]
To explain my problem I added the following example (according to the realm standard example):

There are Owners, and Dogs. Dogs do belong to a race and wear collars. They can either be Dalmatiners or poodles and wear a blue or a red collar. An owner can have multiple dogs.

In realm this relation would look like this:

**Owners:**
```java
public class Owner extends RealmObject{
    @PrimaryKey
    private int id;
    
    private RealmList<Dog> dogs;
}
```

**Dogs:**
```java
public class Dog extends RealmObject{
    @PrimaryKey
    private int id;

    private String race; //either "Dalmatiner" or "Poodle"
    private String collarColor; //either "blue" or "red"
}
```

There's OwnerA that owns a poodle with a blue collar and OwnerB that owns a Dalmatiner with a blue collar and a poodle with a red collar.

* OwnerA
    * DogA [race = Poodle, collarColor = blue]
* OwnerB
    * DogB [race = Dalmatiner, collarColor = blue]
    * DogC [race = Poodle, collarColor = red]

Now I'm trying to find all owners that own a poodle with a blue collar, using the following query:

```java
realm.allObjects(Owner.class).where()
    .equalTo("dogs.race", "Poodle")
    .equalTo("dogs.collarColor", "blue")
.findAll();
```

But this query returns OwnerA and OwnerB. OwnerA is obvious because he owns a poodle with a blue collar. This query applies for OwnerB as well because he owns a dog having a blue collar and another dog being a poodle.

A possible workaround would be to extend the dog class to also hold it's owner. Then query for all dogs that match all criteria and get the owners of those dogs. But it wouldn't be possible to do this in a single realm query.

So I suggest to add a function like and() or beginSelectionGroup() (and endSelectionGroup()) to apply multiple conditions on the same member of a RealmList belonging to an RealmObject. The usage of this functions could then look like this for example:

```java
realm.allObjects(Owner.class).where()
    .equalTo("dogs.race", "Poodle")
    .and()
    .equalTo("dogs.collarColor", "blue")
.findAll();
```
or
```java
realm.allObjects(Owner.class).where()
    .beginSelectionGroup()
        .equalTo("dogs.race", "Poodle")
        .equalTo("dogs.collarColor", "blue")
    .endSelectionGroup()
.findAll();
```

**Comment by beeender** [comment]
Thanks a lot for the well formatted issue and sorry for the late reply.

Yes, it is something missed in our query API and we definitely should support this.

**Comment by cmelchior** [comment]
This will get fixed by #607, so closing as a duplicate
